netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -245,6 +269,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
         if (!future.isSuccess()) {
           log.log(Level.WARNING, "Error shutting down server", future.cause());
         }
+        for (Instrumented<SocketStats> listenSocket : listenSockets) {
+          channelz.removeSocket(listenSocket);
```

Comment:
nit: you probably should use the iterator, and remove the sockets as you go along to avoid retaining memory.

This also means the list cannot be Immutable.

Reply:
I don't think there will be enough server listen sockets for this to be a serious issue. I can set the list to null with low effort, however.